modules/rntuple.mjs

```diff
@@ -141,11 +141,28 @@ deserializeFooter(footer_blob) {
 
     const reader = new RBufferReader(footer_blob);
 
-    this.footerFeatureFlags = reader.readU32();
-    this.headerChecksum = reader.readU32();
+    // Read the envelope metadata
+    this._readEnvelopeMetadata(reader);
 
-    console.log('Footer Feature Flags:', this.footerFeatureFlags);
-    console.log('Header Checksum:', this.headerChecksum);
+
+    // Feature flag(32 bits)
+    this._readFeatureFlags(reader);
+    // Header checksum (64-bit xxhash3)
+    this.headerChecksum = reader.readU64(); 
+
+    const schemaExtensionSize = reader.readS64(); 
+
+    console.log('Schema extension frame size:', schemaExtensionSize);
+
+
+    // Schema extension record frame (4 list frames inside)
+    this._readFieldDescriptors(reader);
```

Comment:
The schema extension reading can later be refactored into a separate function since it's the same code as the header (same with the code to read the frames).